### ISSUE TITLE
It's confusing that Replies to your own messages (and highlights in general?) are entirely coloured red.

### ISSUE THREAD

**Issue by ara4n** [element-hq/element-meta#744]
It's probably fine that replies to your own messages get turned into highlights.  But it's not ideal that it makes the whole message red.  And in fact, highlights in general probably shouldn't be entirely red - only the bit which actually highlighted you.

**Comment by mdsina** [comment]
any chance this issue will be fixed?
very annoyed by this red texts =\

**Comment by nhhollander** [comment]
@mdsina I've been tied up with work lately, but I might have some time tonight or later this week to revisit this.

**Comment by blark** [comment]
This is especially annoying when your name is also an english word. I get red messages anytime someone uses the work mark. Is it possible to remove this and only notify when someone does `@name`?

**Comment by Avamander** [comment]
@blark 

Only one I've hear is that you totally disable notifications/pings except for custom keyword that is your full `@user:homeserver.com`

I'm honestly really surprised that the ping method is currently and has remained this rigid and annoying.

**Comment by newdawncrypto** [comment]
> matrix-org/matrix-react-sdk#3553

Can we please get rid of this annoying red warning color in replies? This thread has been open for 4 years now and still nothing has been done. It's such an easy fix and I am convinced some people aren't using Element simply because of this warning color being used. For the sake of increased adoption and ease on all of our eyes, please change the text color to literally anything else legible.

**Comment by turt2live** [comment]
Spamming issues and PRs with similar comments isn't going to make people inclined to fix it, honestly. The team is currently focusing on other areas of the app, but that doesn't mean a PR is forbidden on this: please absolutely open one if you're confident in the change.

**Comment by Avamander** [comment]
@turt2live There is a PR open about this but it's just sitting still.

Seriously **anything** besides the current alert fatigue inducing red would be an improvement over the years long standstill going on.

I don't know if it's the perfect that has become the enemy of good or it has simply fallen through the cracks, but it's seriously silly and can trivially be remediated to a large extent.

**Comment by LudwigArisleib** [comment]
I agree. Almost no one uses the reply function for serious discussions with long replies in the matrix debate rooms I participate, because the long red texts are unpleasant to the eyes.

Wouldn't it be a 2 minutes fix for a developer (just change the RBG value to make the texts white)? It should be an easy fix for wider adoption.

**Comment by t3chguy** [comment]
> Wouldn't it be a 2 minute fix for a developer?

Yup, it will be. Once the Design team issues instruction to do so

**Comment by robintown** [comment]
We do already have design instruction, actually, but https://github.com/matrix-org/matrix-react-sdk/pull/9199 cannot currently be merged, because there is still outstanding work to be done: specifically, highlighting any custom keywords that caused the notification by placing them in red pills, just like @​room and @​user pills look today, except without the avatar. This is non-trivial, so we are very much open to community contributions that would help implement this final piece.

**Comment by K4LCIFER** [comment]
I'm also eagerly awaiting this to be fixed -- the red is nasty.

**Comment by zeta0134** [comment]
Just chiming in with support for toning this style down. I spent an hour just now updating element on all of my devices and checking logs on my server, wondering why some of my messages were indicating an **error** despite apparently being received on all of my other devices, others being able to reply and react to them, etc. 

Turns out this is just how a normal reply looks? That was confusing... Red usually indicates a **problem**!

**Comment by 5ika** [comment]
Hi, mid-2024 here. Still a problem :-/

**Comment by SkyJumper409x** [comment]
haiii
I noticed the red text and honestly was really confused.
Probably because red is used for warnings and "Dangerous Buttons" everywhere else in the app.
I would prefer the color being customizable, like a setting where it can be changed to pink or blue or whatever (ideally you could have a color picker or the option to input an HTML color code.
I dont mind that red+yellow may look ugly, i just think i broke something when i see red text

**Comment by taffyko** [comment]
https://github.com/element-hq/element-meta/issues/744#issuecomment-1440712735
> We do already have design instruction, actually, but [matrix-org/matrix-react-sdk#9199](https://github.com/matrix-org/matrix-react-sdk/pull/9199) cannot currently be merged, because there is still outstanding work to be done: specifically, highlighting any custom keywords that caused the notification by placing them in red pills, just like @​room and @​user pills look today, except without the avatar. This is non-trivial, so we are very much open to community contributions that would help implement this final piece.

I've opened a PR that resolves this issue and implements Robin's suggestion for notification keywords above!
- ~~https://github.com/element-hq/matrix-react-sdk/pull/85~~
- https://github.com/element-hq/element-web/pull/28270
  Updated PR for element-web, as matrix-react-sdk was incorporated and archived on 2024-10-22

| Before                                                                                                          | After                                                                                                           |
| --------------------------------------------------------------------------------------------------------------- | --------------------------------------------------------------------------------------------------------------- |
| ![Pasted image 20240922225036](https://github.com/user-attachments/assets/96de9454-c2d6-48d7-b2f3-66566a0efeda) | ![Pasted image 20240922225435](https://github.com/user-attachments/assets/865e5a46-011a-4f5a-8af8-44620dfb54a1) |
| ![Pasted image 20240922225030](https://github.com/user-attachments/assets/e345c561-0d3b-407e-990c-b22a91ff1655) | ![Pasted image 20240922225105](https://github.com/user-attachments/assets/a94f12d6-fd3e-4fd6-93df-cac6e689607f) |


---
A few words:

***This issue was consistently brought up by every user that I've onboarded on to Element/Matrix.***
I've been waiting for about 5 years for this to be addressed.

There have been [at](https://github.com/element-hq/element-web/issues/441) [least](https://github.com/element-hq/element-web/issues/13963) [five](https://github.com/element-hq/element-web/issues/18802) [different](https://github.com/element-hq/element-ios/issues/7616) [tickets](https://github.com/element-hq/element-web/issues/27942) opened and [four](https://github.com/matrix-org/matrix-react-sdk/pull/2896) [attempted](https://github.com/matrix-org/matrix-react-sdk/pull/5724) [pull](https://github.com/matrix-org/matrix-react-sdk/pull/9199) [requests](https://github.com/matrix-org/matrix-react-sdk/pull/10533) targeting this issue.

It breaks my heart to see such a trivial and immediate source of friction kept alive for 5 years by bikeshedding — I want to be excited about Element, _I want to be able to recommend Matrix to people_, for there to be a mature, user-friendly client that they can comfortably jump into, but things like this make my confidence in the project waver.

A step in the right direction is so much better than no step at all
— I would exchange a glaring UX issue like this for a [minor edge-case in clarity](https://github.com/matrix-org/matrix-react-sdk/pull/9199#issuecomment-1497987388) any day.

Not every issue needs to be taken out in a single swing.
It's okay to roll out what you [have](https://github.com/matrix-org/matrix-react-sdk/pull/2896) [right](https://github.com/matrix-org/matrix-react-sdk/pull/5724) [now](https://github.com/matrix-org/matrix-react-sdk/pull/9199), to alleviate an issue that impacts the impression of virtually all new users _(where stopping the bleeding as early as possible is the priority!)_
— and then follow-up later on the more minor edge-cases that deserve consideration (especially when you already know what the regressions will be and can proactively prepare tickets and design for them!).

_This is a significantly preferable alternative to letting the situation remain unimproved for half a decade._

**Comment by Titaniumtown** [comment]
I always though the red was something to do with encryption failing or something. Bad UX! Hope this can be improved. I like @taffyko's work on the matter and it seems more intuitive. Great job!

**Comment by pxldevv** [comment]
I'm honestly surprised it was left like this for that long. It seems like an easy patch and it threw me off for a while thinking it was insecure to send replies 💀

**Comment by pcroland** [comment]
Issue opened six years ago, still not fixed.
